checker.c

```diff
@@ -1,21 +1,115 @@
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
-}
-
-int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
+int soccheckPOS(float soc);
+int temperaturecheckPOS(temperature);
+
+int temperaturecheckNEG(float temperature)
+{
+    if(temperature < 0 || temperature > 45)
+    {
+        printf("Temperature out of range!\n");
+        return 0;
+    }
+    else
+    {
+        int result = temperaturecheckPOS(temperature);
+        return result;
+    }
+    
+}
+
+int temperaturecheckPOS(temperature)
+{
+    if(temperature >= 0 || temperature <= 45)
+    {
+        return 1;
+    }
+}
+
+int soccheckNEG(float soc)
+{
+    if(soc < 20 || soc > 80)
```

Comment:
This code is very similar to temperature. See if you can make a single abstraction, like `parameterIsInRange`